onnx/numpy_helper.py

```diff
@@ -118,6 +118,131 @@ def from_array(arr, name=None):  # type: (np.ndarray[Any], Optional[Text]) -> Te
     return tensor
 
 
+def to_list(sequence):  # type: (SequenceProto) -> List[Any]
+    """Converts a sequence def to a Python list.
+
+    Inputs:
+        sequence: a SequenceProto object.
+    Returns:
+        lst: the converted list.
+    """
+    lst = []
+    elem_type = sequence.elem_type
+    value_field = mapping.STORAGE_ELEMENT_TYPE_TO_FIELD[elem_type]
+    values = getattr(sequence, value_field)
+    for value in values:
+        if elem_type == Sequence.TENSOR or elem_type == Sequence.SPARSE_TENSOR:
+            lst.append(to_array(value))
+        elif elem_type == Sequence.SEQUENCE:
+            lst.append(to_list(value))
+        elif elem_type == Sequence.MAP:
+            lst.append(to_dict(value))
+        else:
+            raise TypeError("The element type in the input sequence is not supported.")
+    return lst
+
+
+def from_list(lst, name=None):  # type: (List[Any], Optional[Text]) -> SequenceProto
+    """Converts a list into a sequence def.
+
+    Inputs:
+        lst: a Python list
+        name: (optional) the name of the sequence.
+    Returns:
+        sequence: the converted sequence def.
+    """
+    sequence = SequenceProto()
+    if name:
+        sequence.name = name
+    elem_type = type(lst[0])
```

Comment:
Can we extend this to allow creating from empty list? Maybe also add optional dtype parameter such that type can be specified if the list is empty.

Reply:
Addressed!